server_src/constants/state.constants.js

```diff
@@ -130,38 +95,155 @@ const OP_STATE = {
   UnknownState: "Unknown State ()" // Unknown State (...) needs proper parsing
 };
 
+// Handy model to understand state.flags
+const OP_STATE_FLAGS = {
+  operational: "operational",
+  printing: "printing",
+  cancelling: "cancelling",
+  pausing: "pausing",
+  resuming: "resuming",
+  finishing: "finishing",
+  closedOrError: "closedOrError",
+  error: "error",
+  paused: "paused",
+  ready: "ready",
+  sdReady: "sdReady"
+};
+
 // All states of the app. Nice to share between server and client
 const PSTATE = {
   Offline: "Offline",
   GlobalAPIKey: "Global API Key Issue",
   Searching: "Searching...",
-  "Error!": "Error!",
-  "No-API": "No-API",
+  Error: "Error!",
+  NoAPI: "No-API",
   Disconnected: "Disconnected",
   Starting: "Starting",
   Operational: "Operational",
   Paused: "Paused",
   Printing: "Printing",
   Pausing: "Pausing",
   Cancelling: "Cancelling",
-  "Offline after error": "Offline after error",
+  OfflineAfterError: "Offline after error",
   Complete: "Complete",
   Shutdown: "Shutdown",
   Online: "Online"
 };
 
+const OF_STATE_REMAP = {
+  [OP_STATE.Offline]: {
+    state: PSTATE.Disconnected, // hard remap!
+    desc: "Your printer is disconnected"
+  },
+  [OP_STATE.OpeningSerial]: {
+    state: PSTATE.Searching, // Lack of better
+    desc: "Your printer is connecting to serial"
+  },
```

Comment:
Just curious why these are wrapped in an array?

Reply:
Seen down below.

Reply:
This is how you can define an object key when using a variable. Glad you asked! Found this trick half a year ago when doing the same kind of work for the rugby app

Reply:
I do know this, just forgotten thanks for the reminder!